internal/typeinfo/member.go

```diff
@@ -8,59 +8,82 @@ import (
 
 var scannerInterface = reflect.TypeOf((*sql.Scanner)(nil)).Elem()
 
-// Member describes a type that is a child
-// of some other encapsulating type.
-type Member interface {
-	// OuterType returns the outer type in which this member is present.
-	OuterType() reflect.Type
-
-	// MemberName returns this member's name.
-	MemberName() string
-
-	// ValueFromOuter returns the value represented by this
-	// type member within the input outer value.
-	ValueFromOuter(reflect.Value) (reflect.Value, error)
-
-	// GetScanTarget returns a pointer for the target of rows.Scan, and a
-	// ScanProxy reference in the event that we need to coerce that pointer
-	// into a struct field or map key.
-	GetScanTarget(reflect.Value) (any, *ScanProxy, error)
+// ValueLocator specifies how to locate a value in a SQLair argument type.
+type ValueLocator interface {
+	ArgType() reflect.Type
+	String() string
 }
 
+// Input is a locator for a Go value from SQLair input arguments to be used in
+// a SQL query parameter.
+type Input interface {
+	ValueLocator
+	// LocateParams locates the input argument associated with this Input in
+	// the typeToValue map and then returns the Go values within the input
+	// argument that are to be used in a query parameter. An error is returned
+	// if typeToValue does not contain the input argument.
+	LocateParams(typeToValue map[reflect.Type]reflect.Value) ([]reflect.Value, error)
+}
+
+// Output is a locator for a target to scan results to in the SQLair output
+// arguments.
+type Output interface {
+	ValueLocator
+	// LocateScanTarget locates the output argument associated this Output in
+	// typeToValue and returns a pointer to the Go value within the output
+	// argument for rows.Scan, along with a ScanProxy for the cases where the
+	// output argument cannot be scanned into directly.
+	LocateScanTarget(typeToValue map[reflect.Type]reflect.Value) (any, *ScanProxy, error)
+}
+
+// mapKey stores information about where to find a key of a particular map.
 type mapKey struct {
 	name    string
 	mapType reflect.Type
 }
 
-// OuterType returns the reflected type of the map
-// for which this Member implementation is a key.
-func (mk *mapKey) OuterType() reflect.Type {
+// ArgType returns the type of the map the key is located in.
+func (mk *mapKey) ArgType() reflect.Type {
 	return mk.mapType
 }
 
-// MemberName returns the map key.
-func (mk *mapKey) MemberName() string {
-	return mk.name
+// LocateParams locates the map and then the value of the key specified in
+// mapKey from the provided typeToValue map. An error is returned if the map
```

Comment:
What do you mean with `value of the key [...] from the provided typeToValue map`? Should it be `locates the map using typeToValue and then uses gets the value associated with the key ?